### PR TITLE
Feature/sort filter

### PR DESCRIPTION
**Pull request by Taloreff-hls**
### Sort Functionality:
Users can now sort Pokémon based on attributes such as Name (A-Z, Z-A), Attack (High to Low, Low to High), and HP (High to Low, Low to High).
### Filter Functionality:
Users can search for specific Pokémon by name, and to toggle between all pokemons and my pokemons.

Bug Fixes:

Addressed errors and warnings related to unknown props being passed to DOM elements in MUI table.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,16 @@
 import GlobalStyle from "./assets/globalStyles/GlobalStyle";
 import Header from "./cmps/Header";
 import PokemonContent from "./cmps/PokemonContent";
+import { useState } from "react";
 
 function App() {
+  const [selectedCtg, setSelectedCtg] = useState(0);
+
```
Comment (DinFridman) [inline src/App.tsx]: the app should be clean, create a new component and the state management will be there

File: src/cmps/PokemonCard.tsx [hunk 2]
```diff
@@ -24,14 +24,14 @@ const PokemonCardComponent = ({ pokemon, onClick }: PokemonCardProps) => {
           <PowerLevel>
             <Typography
               type="caption"
-              weight={700}
+              $weight={700}
```
Comment (DinFridman) [inline src/cmps/PokemonCard.tsx]: why do you need the $?

File: src/cmps/PokemonContent.tsx [hunk 3]
```diff
@@ -21,17 +33,32 @@ const PokemonContent = () => {
...
+
+  const sortedPokemonData = pokemonService.sort(
+    filteredPokemonData,
+    sortOption
+  );
+
   return (
     <ContentLayout>
-      <Typography weight={500} type="heading-lg" color="#44484C">
+      <Typography $weight={500} type="heading-lg" color={colors.greys[300]}>
```
Comment (DinFridman) [inline src/cmps/PokemonContent.tsx]: $?

File: src/cmps/PokemonTable.tsx [hunk 4]
```diff
@@ -99,9 +100,11 @@ const PokemonTable = ({ pokemons }: PokemonTableProps) => {
                     >{`#${pokemon.id.toString().padStart(4, "0")}`}</StyledTableCell>
                     <StyledTableCell>{pokemon.description}</StyledTableCell>
                     <StyledTableCell>
-                      {pokemon.base.Attack} Power
+                      {pokemon.base?.Attack ?? "N/A"} Power
```
Comment (DinFridman) [inline src/cmps/PokemonTable.tsx]: move ‘N/A’ to a constant so it can be easily changed everywhere

File: src/cmps/PokemonTable.tsx [hunk 5]
```diff
@@ -121,25 +124,25 @@ const PokemonTable = ({ pokemons }: PokemonTableProps) => {
               </StyledTableRow>
             )}
           </TableBody>
+          <TableFooter>
+            <TableRow>
+              <StyledTablePagination
+                rowsPerPageOptions={[10, 25, 100]}
```
Comment (DinFridman) [inline src/cmps/PokemonTable.tsx]: create const

**Review by DinFridman — CHANGES_REQUESTED** [review]

File: src/cmps/PokemonCard.tsx (hunk 2 above)
Comment (Taloreff-hls) [reply src/cmps/PokemonCard.tsx]: Had a warning in the console: React does not recognize the `weight` prop on a DOM element.

**Review by DinFridman — APPROVED** [review]

**Pull request merged**